File: src/etch/bindings/java/msg/Validator_string.java
package etch.bindings.java.msg;

import etch.bindings.java.transport.fmt.TypeCode;

/**
 * Validator for string.
 */
public class Validator_string extends TypeValidator
{
	/**
	 * @param nDims the number of dimensions. 0 for a scalar.
	 * @return an instance of the validator.
	 */
	public static Validator_string get( int nDims )
    {
		if (nDims < 0)
			return null;
		
		if (nDims >= validators.length)
			return new Validator_string( nDims );
		
		if (validators[nDims] == null)
			validators[nDims] = new Validator_string( nDims );
		
		return validators[nDims];
    }
	
	private static Validator_string[] validators = new Validator_string[4];
	
	/**
	 * Constructs the Validator.
	 *
	 * @param nDims the number of dimensions. 0 for a scalar.
	 */
	private Validator_string( int nDims )
	{
		super( String.class, String.class, nDims, "string["+nDims+"]" );
	}

	@Override
    public Validator elementValidator()
    {
	    return get( nDims-1 );
    }

	@Override
	public Byte checkValue( Object value )
	{
		if (!validate( value))
			return null;
		
		if (nDims > 0)
			return TypeCode.ARRAY;
		
		String v = (String) value;
		
		return v.length() > 0 ? TypeCode.STRING : TypeCode.EMPTY_STRING;
	}
}

File: src/etch/bindings/csharp/msg/Validator_string.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Etch.Msg
{
    /// <summary>
    /// Validator for strings
    /// </summary>
    public class Validator_string : TypeValidator
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="nDims">number of dimensions. 0 for a scalar.</param>
        /// <returns>an instance of the validator</returns>
        public static Validator_string Get( int nDims )
        {
            CheckDims( nDims );

            if ( nDims >= validators.Length )
                return new Validator_string( nDims );

            Validator_string v = validators[ nDims ];

            if ( v == null )
                v = validators[ nDims ] = new Validator_string( nDims );

            return v;
        }

        private static Validator_string[] validators = new Validator_string[ MAX_CACHED ];

        /// <summary>
        /// Constructs the validator.
        /// </summary>
        /// <param name="nDims"></param>
        private Validator_string( int nDims )
            : base( typeof( string ), typeof( string ), nDims, "string["+nDims+"]" )
        { 
        }

        public override Validator ElementValidator()
        {
            return Get( nDims - 1 );
        }

        public override sbyte? CheckValue( object value )
        {
            if ( !Validate( value ) )
                return null;

            if ( nDims > 0 )
                return Etch.Transport.Fmt.TypeCode.ARRAY;

            string v = ( string ) value;

            return v.Length > 0 ? Etch.Transport.Fmt.TypeCode.STRING : Etch.Transport.Fmt.TypeCode.EMPTY_STRING;
        }
    }
}
